src/PhpSpec/CodeGenerator/Generator/MethodGenerator.php

```diff
@@ -111,4 +113,27 @@ protected function getTemplate()
     {
         return file_get_contents(__DIR__.'/templates/method.template');
     }
+
+    /**
+     * @param string $methodName
+     * @param string $snippetToInsert
+     * @param string $code
+     * @return string
+     */
+    private function getUpdatedCode($methodName, $snippetToInsert, $code)
+    {
+        if ('__construct' === $methodName && $this->codeContainsAFunction($code)) {
+            return preg_replace(self::CONSTRUCTOR_PLACEMENT, rtrim($snippetToInsert)."\n\n", $code, 1);
+        }
+        return preg_replace(self::METHOD_PLACEMENT, rtrim($snippetToInsert)."\n}\n", trim($code));
+    }
+
+    /**
+     * @param $code
+     * @return bool
+     */
+    private function codeContainsAFunction($code)
+    {
+        return false !== strpos($code, 'function');
```

Comment:
Regex still isn't that safe; we should probably use the tokenizer more for this stuff